app/celery/v3/notification_tasks.py

```diff
@@ -101,35 +103,41 @@ def v3_process_notification(request_data: dict, service_id: str, api_key_id: str
             template = reader_session.execute(query).one().Template
             notification.template_version = template.version
         except (MultipleResultsFound, NoResultFound):
+            notification.status = NOTIFICATION_PERMANENT_FAILURE
             notification.status_reason = "The template does not exist."
-            # TODO - This isn't an option right now because Notification.template_id is non-nullable and
-            # must reference a valid template.
-            # db.session.add(notification)
-            # db.session.commit()
-            # TODO - Delete logging when the above issue is remedied, and call v3_persist_failed_notification.
-            current_app.logger.error(
-                "Notification %s specified nonexistent template %s.", notification.id, notification.template_id
+            v3_persist_failed_notification(
+                notification,
+                f"Notification {notification.id} specified nonexistent template {notification.template_id}."
             )
             return
 
     notification.template_version = template.version
     if service_id != template.service_id:
+        notification.status = NOTIFICATION_PERMANENT_FAILURE
+        notification.status_reason = "The service does not own the template."
         v3_persist_failed_notification(
-            notification, NOTIFICATION_PERMANENT_FAILURE, "The service does not own the template."
+            notification,
+            f"The service with ID '{service_id}' does not own the template with service ID '{template.service_id}'"
```

Comment:
Multiple templates are likely to have a given service ID.  A better error message is, "Service {service_id} doesn't own template {template_id}."

Reply:
done